pkg/dependencies/factory.go

```diff
@@ -423,7 +423,7 @@ func (f *Factory) WithHelm() *Factory {
 
 func (f *Factory) WithNetworking(clusterConfig *v1alpha1.Cluster) *Factory {
 	var networkingBuilder func() clustermanager.Networking
-	if clusterConfig.Spec.ClusterNetwork.CNI == v1alpha1.Kindnetd {
+	if clusterConfig.Spec.ClusterNetwork.CNI == v1alpha1.Kindnetd || (clusterConfig.Spec.ClusterNetwork.CNIConfig != nil && clusterConfig.Spec.ClusterNetwork.CNIConfig.Kindnetd != nil) {
```

Comment:
I think we can assume defaults have been set and just check `CNIConfig`
Imo that's the advantage of setting defaults, we can rely on that in the rest of the code, just ignoring the old field